packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -605,3 +637,15 @@ function isVirtualPointerEvent(event: PointerEvent) {
     )
   );
 }
+
+function findGenericAncestor(dragTarget: DragTarget, validDropTargets: DropTarget[]): HTMLElement {
+  let ancestor = dragTarget.element.parentElement.closest(':not([aria-hidden]):not([role])');
+  let genericAncestor;
+  while (ancestor && ancestor !== document.body) {
+    if (validDropTargets.every(dropTarget => ancestor.contains(dropTarget.element))) {
+      genericAncestor = ancestor;
+    }
+    ancestor = ancestor.parentElement.closest(':not([aria-hidden]):not([role])');
+  }
+  return genericAncestor;
+}
```

Comment:
I think this could return a generic ancestor that is inside an aria-hidden parent, rather than having aria-hidden on itself. It also would never return `document.body`, which might be the nearest common ancestor in some cases (e.g. portals).

Here's an alternative approach, which should also be a little faster because it doesn't need to traverse upward every iteration (`ancestor.contains`). It collects all ancestors first, and then starts from document.body going inwards until one of the ancestors doesn't match or an `aria-hidden` ancestor is seen. It'll return `document.body` if that's the outermost ancestor, which will work in Chrome but not Firefox.

```js
function findGenericAncestor(dragTarget: DragTarget, validDropTargets: DropTarget[]): HTMLElement {
  // Walk upward from each drop target to get all ancestors.
  let ancestors = [
    getAncestors(dragTarget.element),
    ...validDropTargets.map(dropTarget => getAncestors(dropTarget.element))
  ];

  let minLength = ancestors.reduce((min, parents) => Math.min(min, parents.length), Infinity);

  // Start from outer most parent (document.body) and work inward until one of them doesn't match.
  let last;
  for (let i = 0; i < minLength; i++) {
    let parent = ancestors[0][i];
    if (parent.hasAttribute('aria-hidden') || ancestors.some((p) => p[i] !== parent)) {
      break;
    }

    if (!parent.hasAttribute('role')) {
      last = parent;
    }
  }

  return last;
}

function getAncestors(element: Element) {
  let parents = [];
  while (element !== document.body && element.parentElement) {
    element = element.parentElement;
    parents.unshift(element);
  }
  return parents;
}
```